src/builtins.ts

```diff
@@ -771,6 +800,45 @@ export function compileCall(
         }
       }
     }
+    case BuiltinSymbols.nameof: {
+      // Check to make sure a parameter or a type was passed to the builtin
+      let resultType = evaluateConstantType(compiler, typeArguments, operands, reportNode);
+      if (resultType === null) return module.unreachable();
+
+      let value: string;
+      if (resultType.is(TypeFlags.REFERENCE)) {
+        let classReference = resultType.classReference;
+        if (classReference != null) {
```

Comment:
nit: can skip `!= null`

Reply:
Class reference can be `null`, and should result in a `"Function"` string and a `signatureReference` assertion, right?

I fixed that to this:

 ```ts
if (resultType.is(TypeFlags.REFERENCE)) {
  let classReference = resultType.classReference;
  if (!classReference) {
    assert(resultType.signatureReference);
    value = "Function";
  } else {
    value = classReference.name;
  }
} else {
````